SingularityUI/app/controllers/TaskDetail.coffee

```diff
@@ -89,10 +98,17 @@ class TaskDetailController extends Controller
                 app.caughtError()
                 app.router.notFound()
 
-        @models.resourceUsage?.fetch().error =>
-            # If this 404s there's nothing to get so don't bother
-            app.caughtError()
-            delete @models.resourceUsage
+
+        @models.resourceUsage?.fetch()
+            .done =>
+                # Store current resource usage to compare against future resource usage
+                @models.resourceUsage.setCpuUsage() if @models.resourceUsage.get('previousUsage')              
+                @models.resourceUsage.set('previousUsage', @models.resourceUsage.toJSON())
+
+            .error =>
+                # If this 404s there's nothing to get so don't bother
+                app.caughtError()
+                delete @models.resourceUsage
```

Comment:
could we be a little more clever here and only fetch resource usage if the task hasn't hit a terminal state (`TASK_KILLED`, `TASK_FINISHED`, `TASK_LOST`, etc.)?